File: client/src/app/_models/userParams.ts
// Export this as a class and this way we can give it some initial parameters that we can use. When we
// instantiate new instance of this class.
import {User} from "./user";

export class UserParams {
  gender!: string
  minAge = 18
  maxAge = 99
  pageNumber = 1
  pageSize = 3
  orderBy = 'lastActive'

  constructor(user: User) {
    this.gender = user.gender === 'female' ? 'male' : 'female'
  }
}

File: API/Helpers/UserParams.cs
namespace API.Helpers
{
    // Create a class so that we can receive the pagination parameters from the user.
    public class UserParams : PaginationParams
    {   
        // Add couples properties. 
        public string? CurrentUsername { get; set;}
        public string? Gender { get; set;}
        public int MinAge { get; set; } = 18;    
        public int MaxAge { get; set; } = 150;
        public string? OrderBy { get; set; } = "lastActive";   


    }
}
